### ISSUE TITLE
Command "fmt" is deprecated - Declare a formatting transformer in your kustomization instead

### ISSUE THREAD

**Issue by whyvez** [kubernetes-sigs/kustomize#4339]
When using `kustomize cfg fmt` I got this warning. 

```
Command "fmt" is deprecated, imperative formatting will no longer be available in kustomize v5.
Declare a formatting transformer in your kustomization instead.
```

I don't understand the suggestion, "Declare a formatting transformer in your kustomization instead.".

Can someone explain what this means?

**Comment by natasha41575** [comment]
Apologies for the cryptic and confusing warning. We have decided that commands such as `kustomize cfg fmt` are out of scope for builtin kustomize and would encourage you to instead create a plugin transformer to format your resources instead. You can follow the guides here: https://kubectl.docs.kubernetes.io/guides/extending_kustomize/ to create a plugin transformer, add the formatting transformer to the `transformers` field of your kustomization file, and run `kustomize build` to invoke it.

We would appreciate feedback if either:

- It is difficult to create your own extension for kustomize to format your resources
- This feature is popular enough for us to create a formatting transformer under an official CNCF plugin repo.

Please comment back if you have any further questions!

/kind support
/triage accepted

**Comment by greenkiwi** [comment]
@natasha41575 we just ran across this issue and were wondering whether there were any good examples to follow.  We have attempted to have all of our "code" use strong formatting.  Google-java-formatter for java, black for python, etc.   This has improved our code review as there are fewer _style_ discussions and users are less likely to miss or change large chunks of code.    For k8s objects, having a standard order makes looking at and reviewing diffs much easier.  

We don't really want to build our whole system around a feature that will be removed in kustomize v5, but the alternative seems to be pretty heavy-handed.  Needing to add a new formatter argument to every `kustomization.yaml` file.  It also seems a bit strange for users to run `kustomize build` to have their source formatted.

Cheer,
Adam
